plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -349,4 +352,66 @@ abstract class MavenPublishBaseExtension(
 
     platform.configure(project)
   }
+
+  /**
+   * Calls [configure] with a [Platform] chosen based on other applied Gradle plugins.
+   */
+  @Incubating
+  fun configureBasedOnAppliedPlugins() {
```

Comment:
We could allow optionally passing `javadocJar: JavadocJar? = null, sourcedJar: Boolean = true` (null = we use our current logic) here to eliminate many use cases of calling `configure`.  Only weird thing about this would be that some platforms would ignore the option if it doesn't apply or is not configurable for it.